src/dynamics/mpas/dyn_grid.F90

```diff
@@ -1,13 +1,457 @@
+            1, nedges_solve, lonedge * rad_to_deg, map=global_grid_index)
+
+        allocate(global_grid_map(3, nedges_solve))
+
+        do i = 1, nedges_solve
+            global_grid_map(1, i) = int(i, kind_imap)
+            global_grid_map(2, i) = int(1, kind_imap)
+            global_grid_map(3, i) = global_grid_index(i)
+        end do
+
+        call dyn_debug_print('Registering grid "mpas_edge" with id ' // stringify([dyn_grid_id('mpas_edge')]))
+
+        call cam_grid_register('mpas_edge', dyn_grid_id('mpas_edge'), lat_coord, lon_coord, global_grid_map, &
+            unstruct=.true., block_indexed=.false.)
+
+        deallocate(global_grid_index)
+        nullify(global_grid_index)
+        nullify(global_grid_map)
+        nullify(lat_coord)
+        nullify(lon_coord)
+
+        ! Construct coordinate and grid objects for vertex node grid (i.e., "mpas_vertex").
+        ! Standard MPAS coordinate and dimension names are used.
+
+        call mpas_dynamical_core % get_variable_value(indextovertexid, 'mesh', 'indexToVertexID')
+        call mpas_dynamical_core % get_variable_value(latvertex, 'mesh', 'latVertex')
+        call mpas_dynamical_core % get_variable_value(lonvertex, 'mesh', 'lonVertex')
+
+        allocate(global_grid_index(nvertices_solve))
+
+        global_grid_index(:) = int(indextovertexid(1:nvertices_solve), kind_imap)
+
+        lat_coord => horiz_coord_create('latVertex', 'nVertices', nvertices_global, 'latitude', 'degrees_north', &
+            1, nvertices_solve, latvertex * rad_to_deg, map=global_grid_index)
+        lon_coord => horiz_coord_create('lonVertex', 'nVertices', nvertices_global, 'longitude', 'degrees_east', &
+            1, nvertices_solve, lonvertex * rad_to_deg, map=global_grid_index)
+
+        allocate(global_grid_map(3, nvertices_solve))
+
+        do i = 1, nvertices_solve
+           global_grid_map(1, i) = int(i, kind_imap)
+           global_grid_map(2, i) = int(1, kind_imap)
+           global_grid_map(3, i) = global_grid_index(i)
+        end do
+
+        call dyn_debug_print('Registering grid "mpas_vertex" with id ' // stringify([dyn_grid_id('mpas_vertex')]))
+
+        call cam_grid_register('mpas_vertex', dyn_grid_id('mpas_vertex'), lat_coord, lon_coord, global_grid_map, &
+            unstruct=.true., block_indexed=.false.)
+
+        deallocate(global_grid_index)
+        nullify(global_grid_index)
+        nullify(global_grid_map)
+        nullify(lat_coord)
+        nullify(lon_coord)
+    end subroutine define_cam_grid
+
+    !> Helper function for returning grid id given its name.
+    !> (KCW, 2024-03-27)
+    pure function dyn_grid_id(name)
+        character(*), intent(in) :: name
+        integer :: dyn_grid_id
+
+        integer, parameter :: dyn_grid_id_offset = 0
+        integer :: i
+
+        do i = 1, size(dyn_grid_name)
+            if (trim(adjustl(dyn_grid_name(i))) == trim(adjustl(name))) then
+                dyn_grid_id = dyn_grid_id_offset + i
```

Comment:
I don't think we really need `dyn_grid_id_offset` here:

```suggestion
        integer :: i

        do i = 1, size(dyn_grid_name)
            if (trim(adjustl(dyn_grid_name(i))) == trim(adjustl(name))) then
                dyn_grid_id = i
```

Reply:
The rationale behind the integer constant `dyn_grid_id_offset` here is that the physics grid is registered with a grid id of 100 [[1]](https://github.com/ESCOMP/CAM-SIMA/blob/a097e4e270f9ad071fbe07eaf25a4b91640fa74b/src/physics/utils/physics_grid.F90#L54)[[2]](https://github.com/ESCOMP/CAM-SIMA/blob/a097e4e270f9ad071fbe07eaf25a4b91640fa74b/src/physics/utils/physics_grid.F90#L315).

And in SE dycore, it follows the numbering and the grid ids count from 101 [[3]](https://github.com/ESCOMP/CAM-SIMA/blob/a097e4e270f9ad071fbe07eaf25a4b91640fa74b/src/dynamics/se/dyn_grid.F90#L59). If MPAS dycore were to follow this convention, we can just set `dyn_grid_id_offset` to 100, and the pure function `dyn_grid_id` will give us grid ids that count from 101.

Is this functionality desirable? If not, I can remove it like you suggested.

Reply:
That functionality totally makes sense to me, but looking at the code it appears that `dyn_grid_id_offset` is set to zero (which means it is doing nothing), instead of 100.  So instead of creating a new `dyn_grid_id_offset` parameter I might just use the `phys_decomp` parameter from `phys_grid` and use that here instead, which I think will give you the functionality required (while ensuring that there is no grid ID collision between the physics and dynamics).

Reply:
I pushed https://github.com/ESCOMP/CAM-SIMA/pull/257/commits/d40ff0e972ccf1ea860a02f5dc5f8e38fab753a4 to address this. In addition, the log example in the PR body was updated, showing that dynamics grid ids count from `phys_decomp + 1` now.